wsmaster/che-core-api-factory-bitbucket-server/src/main/java/org/eclipse/che/api/factory/server/bitbucket/HttpBitbucketServerApiClient.java

```diff
@@ -340,6 +323,26 @@ private BitbucketUser getUser(Optional<String> token)
     }
   }
 
+  private String getUsername(HttpResponse<InputStream> response)
+      throws ScmCommunicationException, ScmUnauthorizedException {
+    try {
+      // Try to get the username from the response header.
+      if (response.headers().firstValue(USERNAME_HEADER).isPresent()) {
+        return response.headers().firstValue(USERNAME_HEADER).get();
```

Comment:
are we sure that the header is going to be there ?

```Retrieve version information and other application properties. No authentication is required to call this resource.```

Reply:
I could not find any direct documentation about it, but the only more or less official source I could find, is a discussion in the Atlassian community forum: https://community.developer.atlassian.com/t/obtain-authorised-users-username-from-api/24422.

Reply:
So yes, there is no official documentation about the `X-AUSERNAME` header except the forum discussion, but on the other hand there is no other way to get the username of the authenticated user.